pkg/metrics/certificates.go

```diff
@@ -107,9 +116,7 @@ func (m *Metrics) RemoveCertificate(key string) {
 		return
 	}
 
-	m.certificateExpiryTimeSeconds.DeleteLabelValues(name, namespace)
-	m.certificateRenewalTimeSeconds.DeleteLabelValues(name, namespace)
-	for _, condition := range readyConditionStatuses {
-		m.certificateReadyStatus.DeleteLabelValues(name, namespace, string(condition))
-	}
+	m.certificateExpiryTimeSeconds.DeletePartialMatch(prometheus.Labels{"name": name, "namespace": namespace})
```

Comment:
Not sure it makes sense for us to be using partial matching if we know all the label values. Can we please change is back to how it was, and pass in the issuer ref values?

Reply:
Hey @JoshVanL, thanks for the feedback! indeed, this was my initial approach, however what I found is that when the metrics are removed here: https://github.com/cert-manager/cert-manager/blob/52c6723de522a63a66ef8ea8d99117f6b8a79b5b/pkg/controller/certificates/metrics/controller.go#L96

the `crt` object is already gone, since the conditional for the removal is when the crt could not be found:
https://github.com/cert-manager/cert-manager/blob/52c6723de522a63a66ef8ea8d99117f6b8a79b5b/pkg/controller/certificates/metrics/controller.go#L94

so I couldn't retrieve the IssuerRef attributes at this point (otherwise get `Invalid memory address or nil pointer dereference`)

I tried to figure out a way to propagate this somehow, but seemed to me like it would over complicate things considerably and since the `key` is uniquely identifying all the time series the should be deleted using `DeletePartialMatch` seemed appropriate.

Reply:
Understood, thanks @dkulchinsky! This makes sense to me.

/lgtm
/ok-to-test
/approve

Reply:
thank @JoshVanL! happy to see this getting merged 😄 now can't wait for next release 🤣